homeassistant/components/switch/broadlink.py

```diff
@@ -115,23 +121,28 @@ def _learn_command(call):
     def _send_packet(call):
         """Send a packet."""
         packets = call.data.get('packet', [])
-        for packet in packets:
-            for retry in range(DEFAULT_RETRY):
-                try:
-                    extra = len(packet) % 4
-                    if extra > 0:
-                        packet = packet + ('=' * (4 - extra))
-                    payload = b64decode(packet)
-                    yield from hass.async_add_job(
-                        broadlink_device.send_data, payload)
-                    break
-                except (socket.timeout, ValueError):
+        repeat = call.data.get('repeat')
+        delay = call.data.get('delay')
+        for _ in range(0, repeat):
+            for packet in packets:
+                for retry in range(DEFAULT_RETRY):
                     try:
+                        extra = len(packet) % 4
+                        if extra > 0:
+                            packet = packet + ('=' * (4 - extra))
+                        payload = b64decode(packet)
                         yield from hass.async_add_job(
-                            broadlink_device.auth)
-                    except socket.timeout:
-                        if retry == DEFAULT_RETRY-1:
-                            _LOGGER.error("Failed to send packet to device")
+                            broadlink_device.send_data, payload)
+                        yield from asyncio.sleep(delay)
```

Comment:
We should not sleep after the last command. That's easier to accomplish by keeping track of a `should_delay` boolean that is set to True after a command has been sent and is set to False whenever sleep is done.